### PR TITLE
Split fetch CORP tests as HTTP and HTTPS.

### PR DESCRIPTION
**Pull request by youennf**
Add fetch.https.any.js in addition to fetch.any.js.
Make fetch.any.js not run in service worker.
Make fetch.https.any.js run also in service worker.
Update a test in fetch.any.js for the "same-site HTTP -> HTTPS" case.

### CODE REVIEW

**Comment by anforowicz** [comment]
Drive-by comment... feel free to ignore :-)

Thank you very much for putting together the PR.  I can confirm that the tests pass against my Chromium WIP CL @ https://chromium-review.googlesource.com/c/chromium/src/+/1416813.

And separate thanks for having the luxury of implementing the feature with already existing test coverage .  One benefit is that it prevent me from forgetting to support redirects (the last testcase in fetch.any.js fails).  Thanks!

File: fetch/cross-origin-resource-policy/fetch.any.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ promise_test((test) => {
 
 promise_test((test) => {
     const remoteURL = httpsBaseURL + "resources/hello.py?corp=same-site";
-    return fetch(remoteURL, { mode: "no-cors" });
+    return promise_rejects(test, new TypeError, fetch(remoteURL, { mode: "no-cors" }));
 }, "Cross-origin no-cors fetch to a same-site URL with a 'Cross-Origin-Resource-Policy: same-site' response header.");
```
Comment (annevk) [inline fetch/cross-origin-resource-policy/fetch.any.js]: Please make this say "cross-scheme same-site" or some such. I forgot to read the PR description and it took me a long time to figure out what was going on.

File: fetch/cross-origin-resource-policy/fetch.https.any.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
+// META: global=window,worker
+// META: script=/common/get-host-info.sub.js
+
+const host = get_host_info();
+const path = "/fetch/cross-origin-resource-policy/";
+const localBaseURL = host.HTTPS_ORIGIN + path;
+const notSameSiteBaseURL = host.HTTPS_NOTSAMESITE_ORIGIN + path;
+const httpBaseURL = host.HTTP_ORIGIN + path;
```
Comment (annevk) [inline fetch/cross-origin-resource-policy/fetch.https.any.js]: This appears unused, remove it?

**Review by annevk — APPROVED** [review]
Looks great, two minor nits.

File: fetch/cross-origin-resource-policy/fetch.any.js (hunk 1 above)
Comment (youennf) [reply fetch/cross-origin-resource-policy/fetch.any.js]: OK, added Cross-scheme (HTTP->HTTPS)

**Comment by youennf** [comment]
Taskcluster is failing for both Firefox and Chrome.
Looking at the logs, there does not seem to be anything suspicious related to the PR.
@foolip, any idea what this might be?

**Comment by annevk** [comment]
I think the problem is the change to `common/get-host-info.sub.js` which presumably triggers runs of lots of tests, causing the whole thing to time out.

**Pull request merged**